pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/route/PatchFormAnswersRoute.java

```diff
@@ -762,29 +766,31 @@ private void checkAnswerConstraints(Response response, Answer answer) {
     private void verifyFileUpload(Handle handle, Response response, ActivityInstanceDto instanceDto, FileAnswer answer) {
         long participantId = instanceDto.getParticipantId();
         long studyId = instanceDto.getStudyId();
-        long uploadId = answer.getValue().getUploadId();
-
-        var verifyResult = fileService.verifyUpload(handle, studyId, participantId, uploadId)
-                .orElseThrow(() -> new DDPException("Could not find file upload with id " + uploadId));
-
-        switch (verifyResult) {
-            case NOT_UPLOADED:
-                throw ResponseUtil.haltError(response, 400, new ApiError(ErrorCodes.FILE_ERROR,
-                        "File has not been uploaded yet"));
-            case OWNER_MISMATCH:
-                throw ResponseUtil.haltError(response, 400, new ApiError(ErrorCodes.FILE_ERROR,
-                        "File is not owned by participant and cannot be associated with answer"));
-            case QUARANTINED:
-                throw ResponseUtil.haltError(response, 400, new ApiError(ErrorCodes.FILE_ERROR,
-                        "File is infected and cannot be associated with answer"));
-            case SIZE_MISMATCH:
-                throw ResponseUtil.haltError(response, 400, new ApiError(ErrorCodes.FILE_ERROR,
-                        "File uploaded size does not match expected size"));
-            case OK:
-                LOG.info("File upload with id {} is uploaded and can be associated with participant's answer", uploadId);
-                break;
-            default:
-                throw new DDPException("Unhandled file check result: " + verifyResult);
+        List<FileInfo> fileInfos = answer.getValue();
```

Comment:
Let's break down this method into two separate. Since you're just adding a cycle, you are able to reuse the previously defined method.
For instance:
`verifyFileUpload() {
  for (FileInfo info : fileInfos)
    verifyFileInfo(info);
}`